src/__tests__/indexing.ts

```diff
@@ -38,6 +39,44 @@ const extractDocuments = (
   )
 }
 
+const waitForMidsToBeIndexed = (
+  ceramic: CeramicApi,
+  docs: ModelInstanceDocument[]
+): Promise<Record<string, ModelInstanceDocument>> => {
+  return firstValueFrom(
+    interval(1000).pipe(
+      mergeMap(() => ceramic.index.query({ model: TEST_MODEL, last: 100 })),
+      map(resultObj => extractDocuments(ceramic, resultObj)),
+      map(results =>
```

Comment:
I'm sorry, this broke my brain.
I find this type of complex functional code really hard to follow.  Is there any way to make it easier to follow what this is doing?  Easiest for me would probably be just to write it iteratively instead of functionally, but also maybe comments or breaking it up to smaller helper functions would possibly be another option?

Reply:
yup will do